src/index.rs

```diff
@@ -130,6 +133,17 @@ impl Index {
         heap.into_sorted_vec().iter().map(|e| e.0).collect()
     }
 
+    /// save the index to a file
+    pub fn save(&self, path: &str) {
```

Comment:
I suggest changing this (and the load function) one of two ways:
- either take a file object in the argument, so that what appears to be an independent library doesn't interact with the file system (and its general complexities) directly; or,
- make Index serializable (which you did) and move the implementation of save and load to the upstream script.

Reply:
Implemented first option.